hail_search/queries/mito.py

```diff
@@ -252,15 +273,49 @@ def _apply_multi_sample_type_entry_filters(self, ht, family_idx_map):
         # Filter out families with no valid entries in either sample type
         return ht.filter(ht.family_entries.any(hl.is_defined))
 
+    def _apply_quality_entry_filters(self, ht, sample_type, family_idx_map):
+        return ht.annotate(**{
+            sample_type.family_entries_field: hl.enumerate(ht[sample_type.family_entries_field]).starmap(
+                lambda i, family_samples: hl.or_missing(
+                    hl.bind(lambda other_sample_type_idx: (
+                        self._family_has_valid_quality(ht, sample_type, i) |
+                        self._family_has_valid_quality(ht, sample_type.other_sample_type, other_sample_type_idx)
+                    ), family_idx_map.get(hl.coalesce(family_samples)[0]['familyGuid']).get(sample_type.other_sample_type.value),
+                ), family_samples)
+            )})
+
     @staticmethod
-    def _family_has_valid_sample_type_entries(ht, sample_type, sample_type_family_idx):
-        # Note: This logic does not sufficiently handle case 2 here https://docs.google.com/presentation/d/1hqDV8ulhviUcR5C4PtNUqkCLXKDsc6pccgFVlFmWUAU/edit?usp=sharing
-        # and will need to be changed to support it - https://github.com/broadinstitute/seqr/issues/4403
+    def _family_has_valid_quality(ht, sample_type, sample_type_family_idx):
         return (
             hl.is_defined(sample_type_family_idx) &
-            hl.is_defined(ht[sample_type.passes_quality_field][sample_type_family_idx]) &
-            hl.is_defined(ht[sample_type.passes_inheritance_field][sample_type_family_idx])
+            hl.is_defined(ht[sample_type.passes_quality_field][sample_type_family_idx])
+        )
+
+    @staticmethod
+    def _apply_inheritance_entry_filters(ht, sample_type, family_idx_map):
+        ht = ht.annotate(
+            **{sample_type.family_entries_field: hl.enumerate(ht[sample_type.family_entries_field]).starmap(
+                lambda family_idx, family_samples: hl.or_missing(
+                    hl.bind(lambda other_sample_type_family_idx: (
+                        hl.bind(
+                            lambda other_sample_type_pass_samples, sample_type_pass_samples: (
+                                ht[sample_type.failed_family_sample_field][family_idx].all(
+                                    other_sample_type_pass_samples.contains
+                                ) & ht[sample_type.other_sample_type.failed_family_sample_field][other_sample_type_family_idx].all(
+                                    sample_type_pass_samples.contains
+                                )),
```

Comment:
these binds are a little wild, but the logic is that for all families, keep family if for all samples, keep a failed sample if sample failed in one sample type it passed in the other